erl/erl.go

```diff
@@ -0,0 +1,257 @@
+	// [RateWindow1s], [RateWindow10s], and [RateWindow60s].
+	RateWindow = fastly.RateWindow
+
+	// CounterDuration indicates the estimated number of events in this
+	// duration in the current POP.  Counts are divided into 10 second
+	// buckets, and each bucket represents the estimated number of
+	// requests received up to and including that 10 second window.
+	//
+	// Buckets are not continuous.  For example, if the current time is
+	// 12:01:03, then the 10 second bucket represents events received
+	// between 12:01:00 and 12:01:10, not between 12:00:53 and 12:01:03.
+	// This means that, in each minute at the ten second mark (:00, :10,
+	// :20, etc.) the window represented by each bucket will shift.
+	//
+	// Estimated counts are not precise and should not be used as
+	// counters.
+	//
+	// See [CounterDuration10s], [CounterDuration20s],
+	// [CounterDuration30s], [CounterDuration40s], [CounterDuration50s],
+	// and [CounterDuration60s].
+	CounterDuration = fastly.CounterDuration
+)
+
+// RateCounter is a named counter that can be incremented and queried.
+type RateCounter struct {
+	name string
+}
+
+// OpenRateCounter opens a rate counter with the given name, creating it
+// if it doesn't already exist.  The rate counter name may be up to 64
+// characters long.  Entry names in this counter are also limited to 64
+// characters.
+func OpenRateCounter(name string) *RateCounter {
+	return &RateCounter{name: name}
+}
+
+// Increment increments the rate counter for this entry by the given
+// delta value.  The minimum value is 0 and the maximum is 1000.
+func (rc *RateCounter) Increment(entry string, delta uint32) error {
+	return mapFastlyError(fastly.RateCounterIncrement(rc.name, entry, delta))
+}
+
+// LookupRate returns the rate of events per second over the given rate
+// window for this entry.
+func (rc *RateCounter) LookupRate(entry string, window RateWindow) (uint32, error) {
+	v, err := fastly.RateCounterLookupRate(rc.name, entry, window)
+	if err != nil {
+		return 0, mapFastlyError(err)
+	}
+	return v, nil
+}
+
+// LookupCount returns the estimated number of events in the given
+// duration for this entry.  The duration represents a discrete window,
+// not a continuous one.  See [CounterDuration] for more details.
+func (rc *RateCounter) LookupCount(entry string, duration CounterDuration) (uint32, error) {
+	v, err := fastly.RateCounterLookupCount(rc.name, entry, duration)
+	if err != nil {
+		return 0, mapFastlyError(err)
+	}
+	return v, nil
+}
+
+// PenaltyBox is a type that allows entries to penalized for a given
+// number of minutes in the future.
+type PenaltyBox struct {
+	name string
+}
+
+// OpenPenaltyBox opens a penalty box with the given name, creating it
+// if it doesn't already exist.  The penalty box name may be up to 64
+// characters long.  Entry names in this penalty box are also limited to
+// 64 characters.
+func OpenPenaltyBox(name string) *PenaltyBox {
+	return &PenaltyBox{name: name}
```

Comment:
Worth adding a check like this?
```suggestion
	if len(name) > 64 {
		panic("penalty box: name too long")
	}
	return &PenaltyBox{name: name}
```